sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -207,9 +207,11 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
         className = table.provider.get,
         options = table.storage.properties)
 
-    LogicalRelation(
+    val logicalRel = LogicalRelation(
       dataSource.resolveRelation(),
       metastoreTableIdentifier = Some(table.identifier))
+    logicalRel.inheritedStats = table.catalogStats
```

Comment:
Shouldn't we just add all the table metadata to the LogicalRelation? Then we don't have to assign a statistics.